src/background.js

```diff
@@ -86,3 +104,122 @@ if (isDevelopment) {
+      asset =>
+        asset.browser_download_url.includes(arch) && asset.browser_download_url.includes(platform)
+    )
+
+    if (release) {
+      const releaseUrl = release.browser_download_url
+      const latestReleaseVersion = releaseUrl
+        .match(/[/](\d+[.]){2}(\d+)[/]/)[0]
+        .replace('/', '')
+        .replace('/', '')
+      console.log('Latest release version: ' + latestReleaseVersion)
+      const isRunning = await isWalletRunning()
+
+      if (isRunning) {
+        const runningVersion = getCurrentWitnetVersion()
+        if (latestReleaseVersion !== runningVersion) {
+          // stop wallet process
+        }
+      } else {
+        if (!fs.existsSync(SHEIKAH_PATH)) {
+          fs.mkdirSync(SHEIKAH_PATH)
+        }
+
+        const existVersionFile = fs.existsSync(path.join(SHEIKAH_PATH, 'version.txt'))
+        const isLastestVersion =
+          latestReleaseVersion ===
+          fs.readFileSync(path.join(SHEIKAH_PATH, 'version.txt')).toString()
+
+        if (!existVersionFile || !isLastestVersion) {
+          await downloadWalletRelease(releaseUrl, latestReleaseVersion)
+        }
+        runWallet()
+      }
+    } else {
+      console.log('Your OS is not supported yet')
+    }
+  })
+}
+
+if (isDevelopment) {
+  main()
+}
+
+async function downloadWalletRelease(releaseUrl, version) {
+  return new Promise((resolve, reject) => {
+    axios
+      .get(releaseUrl, { responseType: 'stream' })
+      .then(async response => {
+        const file = `witnet-release-${arch}-${platform}.tar.gz`
+        const pipeline = util.promisify(stream.pipeline)
+        // Promise equivalent for response.data.pipe(writeStream)
+        await pipeline(response.data, fs.createWriteStream(file))
+        console.log('witnet release downloaded succesfully')
+
+        console.log('Decompressing release...')
+        // Decompress tar.gz file
+        tar.x({
+          file,
+          sync: true,
+        })
+        // TODO: tar is not decompressing correctly if a pass a path.
+        // Create these files in ./sheikah
+        fs.copyFileSync('witnet', `${SHEIKAH_PATH}/witnet`)
+        fs.copyFileSync('witnet.toml', `${SHEIKAH_PATH}/witnet.toml`)
+        fs.writeFileSync(path.join(SHEIKAH_PATH, 'version.txt'), version)
+
+        resolve()
+      })
+      .catch(err => {
+        console.log('err', err)
+      })
+  })
+}
+
+async function isWalletRunning() {
+  const processes = await psList()
+  // Run by cargo => cmd: 'target/debug/witnet -c witnet_01.toml --trace wallet server',
+  // Run by ./wintet => cmd: './witnet wallet server',
+  return !!processes
+    .filter(x => x.name === 'witnet')
+    .find(p => p.cmd.includes('/witnet wallet server'))
+}
+
+function runWallet() {
+  console.log('Running wallet...')
+  // Pass { detached: true, stdio: false } and then call .unref() in child_process will make it independent from its parent.
+  // In other words, if sheikah stops the wallet process will keep alive
```

Comment:
Is this a good idea? How do you stop the wallet? This will force users to run `killall witnet` if something goes wrong...